quint/src/runtime/impl/compilerImpl.ts

```diff
@@ -1254,7 +1254,11 @@ export class CompilerVisitor implements IRVisitor {
   }
 
   // compute all { ... } or A.then(B)...then(E) for a chain of actions
-  private chainAllOrThen(actions: Computable[], kind: 'all' | 'then'): Maybe<EvalResult> {
+  private chainAllOrThen(
+    actions: Computable[],
+    kind: 'all' | 'then',
+    actionId: (idx: number) => bigint
```

Comment:
Please document how this is meant to be used and why we need it.

Reply:
I have added a detailed documentation to this private method :)

Reply:
Private doesn't mean we don't need to be able understand what stuff is supposed to do. So thank you for taking time to add a comment.